File: Razvoj mobilnih aplikacija labosi/RecyclerViwe/app/src/main/java/com/filip/myapplication/MainActivity.java
package com.filip.myapplication;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

public class MainActivity extends AppCompatActivity {
    private RecyclerView listRecyclerView;
    private RecyclerView.LayoutManager layoutManager;

    private int[] image = {R.drawable.download, R.drawable.stevejobsheadshot2010crop2, R.drawable.swh1web};

    private RecyclerAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listRecyclerView = (RecyclerView)findViewById(R.id.recylerViweID);
        layoutManager = new LinearLayoutManager(this);
        listRecyclerView.setLayoutManager(layoutManager);

        adapter = new RecyclerAdapter(image);
        listRecyclerView.setAdapter(adapter);
    }
}

File: Razvoj mobilnih aplikacija labosi/BooksRecycler/app/src/main/java/com/filip/booksrecycler/MainActivity.kt
package com.filip.booksrecycler

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setUpUi()
    }

    private fun setUpUi() {
        bookDisplay.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
        bookDisplay.itemAnimator = DefaultItemAnimator()
        bookDisplay.addItemDecoration(DividerItemDecoration(this,RecyclerView.VERTICAL))
        displayData()
    }
    private fun displayData() {
        val detailsIntent: Intent = Intent(this, DetailsActivity::class.java)
        val bookListener = object: BookInteractionListener {
            override fun onShowDetails(id: Int) {
                when(id){
                    1-> startActivity(detailsIntent)
                }
                val book = BookRepository.get(id)
                Toast.makeText(applicationContext, book.toString(), Toast.LENGTH_SHORT).show()
            }
        }
        bookDisplay.adapter = BookAdapter(BookRepository.books, bookListener)
    }

}

